src/utils/exchange.ts

```diff
@@ -0,0 +1,66 @@
+import BN from 'bn.js'
+import { AuctionElement } from 'api/exchange/ExchangeApi'
+
+const ADDRESS_WIDTH = 20 * 2
+const UINT256_WIDTH = 32 * 2
+const UINT16_WIDTH = 2 * 2
+const UINT32_WIDTH = 4 * 2
+const UINT128_WIDTH = 16 * 2
+
+const hexPattern = '[0-9a-fA-F]'
+const hn = (n: number): string => hexPattern + `{${n}}`
+
+// /(address)(sellTokenBalance)(buyTokenId)(sellTokenId)(validFrom)(validUntil)(priceNumerator)(priceDenominator)(remainingAmount)/g
+const orderPattern = `(?<user>${hn(ADDRESS_WIDTH)})(?<sellTokenBalance>${hn(UINT256_WIDTH)})(?<buyTokenId>${hn(
```

Comment:
I think the artifacts here belongs in the ExchangeApi or closer (in the `api/exchange` folder) rather than utils.
Not like anything else will need this, and the API depends on this to decode user orders.

Reply:
Good point, I'll move it closer

Reply:
Addressed: c2c148dfdc97fbb4e201890b02377daa887e09ea